test/core/client_config/uri_parser_test.c

```diff
@@ -56,6 +56,56 @@ static void test_fails(const char *uri_text) {
   GPR_ASSERT(NULL == grpc_uri_parse(uri_text, 0));
 }
 
+static void test_query_parts() {
+  {
+    const char *uri_text = "http://foo/path?a&b=B&c=&#frag";
```

Comment:
woops, that belongs on a different PR that's already merged (#6034). I'll create a new PR for that :)

Those changes where in here because I simply merged that other branch in order to make progress in parallel.